### PR TITLE
Polish README a bit and remove format.sh from git index

### PR DESCRIPTION
**Pull request by ray6080**
+ Remove `format.sh` from git index since it is in `.gitignore` file, and should no longer be tracked.
+ Polish README a little bit.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/71)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ## Ti-Client in Java [Under Construction]
 
-NOTES: For now it works with only TiKV branch disksing/grpc instead of official release.
+NOTES: For now it works only with the disksing/grpc branch of TiKV instead of the official release one.
```
Comment (zhexuany) [inline README.md]: You can delete the whole line. `TiKV` master is ready.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 A Java client for [TiDB](https://github.com/pingcap/tidb)/[TiKV](https://github.com/pingcap/tikv). 
-It suppose to:
+It is supposed to:
 + Communicate via [gRPC](http://www.grpc.io/)
 + Talk to Placement Driver searching for a region
-+ Talk to TiKV reading / writing data the format/encoding as a real TiDB.
-+ Talk Coprocessor for calculation pushdown
++ Talk to TiKV reading/writing data with the format/encoding as a real TiDB.
```
Comment (zhexuany) [inline README.md]: "Take to TiKV for reading/writing data. The resulted data will be decoding/encoding just like what we do in TiDB. " is better.

File: README.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ make test
 ```
 
 this project is designed to hook with `pd` and `tikv` which you can find in `PingCap` github page.
-For the sake of saving your and our time, we submoudle these already. The following command can download them.
+For the sake of saving your and our time, we've submoudled these already. Run the following command to download them.
```
Comment (zhexuany) [inline README.md]: delete this line. We planed to remove `pd` and `tikv` submodule.

File: scripts/format.sh [hunk 4]
```diff
@@ -1,16 +0,0 @@
-#!/bin/sh
```
Comment (zhexuany) [inline scripts/format.sh]: We add this for formatting. Any reason to remove this?

Comment (ray6080) [reply scripts/format.sh]: Just I see it in `.gitignore` file, so I think it's no longer trackable.

Comment (ray6080) [reply scripts/format.sh]: Maybe should remove it form `.gitignore` ...

File: README.md (hunk 2 above)
Comment (ray6080) [reply README.md]: How about this? "Talk to TiKV for reading/writing data and the resulted data is encoded/decoded just like what we do in TiDB"

File: scripts/format.sh (hunk 4 above)
Comment (zhexuany) [reply scripts/format.sh]: Ah. That is it. I added `format.sh` it with `-f`. remove ignore from `.gitignore` is great.

Comment (ray6080) [reply scripts/format.sh]: OK. I added `format.sh` back and removed it from the `.gitignore`.

File: README.md [hunk 5]
```diff
@@ -16,7 +16,7 @@ The following command can install dependencies for you.
 mvn package
 ```
```
Comment (wolfstudy) [inline README.md:16]: when  execute `mvn package`  , first need to install `maven` on your computer  whether need to give a tip?

**Pull request merged**